idx/memory/memory.go

```diff
@@ -639,42 +597,75 @@ func (m *MemoryIdx) Prune(orgId int, oldest time.Time) ([]idx.Archive, error) {
 	return pruned, nil
 }
 
-// filepath.Match doesn't support {} because that's not posix, it's a bashism
-// the easiest way of implementing this extra feature is just expanding single queries
-// that contain these queries into multiple queries, who will be checked separately
-// and whose results will be ORed.
-func expandQueries(query string) []string {
-	queries := []string{query}
+func getMatcher(path string) (func([]string) []string, error) {
+	// Matches everything
+	if path == "*" {
+		return func(children []string) []string {
+			log.Debug("memory-idx: Matching all children")
+			return children
+		}, nil
+	}
+
+	// Convert to regex and match
+	if strings.ContainsAny(path, "*{}[]?") {
+		r, err := regexp.Compile(toRegexp(path))
+		if err != nil {
+			log.Debug("memory-idx: regexp failed to compile. %s - %s", path, err)
+			return nil, err
+		}
+		return func(children []string) []string {
+			matches := make([]string, 0)
+			for _, c := range children {
+				if r.MatchString(c) {
+					log.Debug("memory-idx: %s regex matches %s", c, r.String())
+					matches = append(matches, c)
+				}
+			}
+			return matches
+		}, nil
+	}
+
+	// Match a particular value
+	return func(children []string) []string {
+		for _, c := range children {
+			if c == path {
+				log.Debug("memory-idx: %s matches %s", c, path)
+				return []string{c}
+			}
+		}
+		return []string{}
```

Comment:
no need to allocate. can just use return `nil` here (which is a valid value of a zero-sized slice. see https://dave.cheney.net/2013/01/19/what-is-the-zero-value-and-why-is-it-useful for example). will speed up cases without matches.